core/arch/arm/mm/sp_mem.c

```diff
@@ -126,12 +125,25 @@ void  sp_mem_set_cattr(struct mobj *mobj, uint32_t cache_type)
 	m->cache_type = cache_type;
 }
 
+void  sp_mem_set_secure(struct mobj *mobj, bool is_secure)
+{
+	struct mobj_sp *m = to_mobj_sp(mobj);
+
+	m->is_secure = is_secure;
+}
+
 static bool mobj_sp_matches(struct mobj *mobj __maybe_unused,
 			    enum buf_is_attr attr)
 {
+	struct mobj_sp *m = to_mobj_sp(mobj);
+
 	assert(mobj->ops == &mobj_sp_ops);
```

Comment:
This assert can be removed, it's done inside `to_mobj_sp()`.

Reply:
Ack